### ISSUE TITLE
Is name in Element mandatory or optional?

### ISSUE THREAD

**Issue by zvr** [spdx/spec-v3-template#5]
In the Core namespace, the diagram states name is [0..1] but Sean's input has it as 1..1.

**Comment by goneall** [comment]
I recall a discussion during 2.0 where we wanted the name to be optional - the example being used was code snippets don't typically have a name associated with them.

**Comment by davaya** [comment]
What is the use case for name?

```
  id: mandatory: Y  unique: Y  opaque: Y  descriptive: Optional
name: mandatory: ?  unique: ?  opaque: Y  descriptive: Y
```
William says that tools may choose whether to include descriptive information in ids.  But since id is opaque, tools are not allowed to take any action/decision based on the content of id.

The _raison d'etre_ of name is to carry a human-meaningful moniker for the element.  The question of whether it is mandatory is answered by deciding whether it is also an id (unique).  If name is required to be a unique primary key then id is redundant and can be discarded.

I believe it is unnecessarily complex and confusing to conflate the purposes of id and name.  My driver's license has an id and a name; the purpose of each is unambiguous.  I believe that in all cases (such as references) where human-readable names make SBOMs easier for humans to play with, names should be included in the class definition and used for that purpose.

My recommendation:
1) for Element name is mandatory and not required to be unique.
2) for classes like Reference, id is mandatory and name is optional.

**Comment by davaya** [comment]
TL;DR:  name should be [0..1]

What is the use case for name?

```
  id: mandatory: Y  unique: Y  opaque: Y  descriptive: Optional
name: mandatory: ?  unique: ?  opaque: Y  descriptive: Y
```
William says that tools may choose whether to include descriptive information in ids.  And since id is opaque, tools are not allowed to take any action/decision conditioned on the content of id.

The _raison d'etre_ of name is to carry a human-meaningful moniker for the element.  The question of whether it is mandatory is partially answered by deciding whether it is also an id (unique).  If name is required to be a unique primary key then name is also mandatory and id is redundant and can be discarded.

I believe it is unnecessarily complex and confusing to conflate the purposes of id and name.  My driver's license has an id and a name; the purpose of each is unambiguous.  I believe that in all cases (such as references to Element) where human-readable names make serialized SBOM data easier for humans to play with, names should be included in the class definition and used for that purpose.

My recommendation:
1) for Element, name is not required to be unique.  For Gary's use case it is also not mandatory, though it would always be populated where a human-readable label is desired.
2) for properties of classes like Relationship, id is mandatory and name can be added at the tool's discretion.  The id and name properties may be concatenated into a single formatted string (e.g., \<id\>[-\<name\>]) when serialized, and unambiguously split into id and name properties when deserialized.

(I haven't seen a use case for relationships to have relationships, but if they can, that's another reason for Element name to be optional.)
